cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -1322,6 +1394,23 @@ func (s *Supervisor) runAgentProcess() {
 				s.telemetrySettings.Logger.Error("Could not stop agent process", zap.Error(err))
 			}
 			return
+
+		case stopRequest := <-s.agentStopChan:
+			err := s.commander.Stop(context.Background())
+			if err != nil {
+				s.telemetrySettings.Logger.Error("Could not stop agent process", zap.Error(err))
+			}
+
+			close(stopRequest.processStopped)
+
+			select {
+			case <-stopRequest.start:
+				err := s.commander.Start(context.Background())
+				if err != nil {
+					s.telemetrySettings.Logger.Error("Could not start agent process", zap.Error(err))
```

Comment:
If we get here should start the restartTimer to give us another chance to get the agent running?